packages/cli/src/commands/deploy/action.ts

```diff
@@ -4,82 +4,119 @@ import Session from '../../models/network/Session';
 import { compile } from '../../models/compiler/Compiler';
 import { fromContractFullName } from '../../utils/naming';
 import NetworkController from '../../models/network/NetworkController';
+import { ProxyType } from '../../scripts/interfaces';
+import ContractManager from '../../models/local/ContractManager';
 import stdout from '../../utils/stdout';
 import { parseMultipleArgs } from '../../utils/input';
-import { createAction } from '../create';
 import { Loggy } from '@openzeppelin/upgrades';
 
+import link from '../link';
+import add from '../add';
+import push from '../push';
+
 import { Options, Args } from './spec';
 
-export async function preAction(params: Options & Args): Promise<void | (() => Promise<void>)> {
-  if (!params.skipCompile) {
-    await compile();
-  }
+function isProxyKind(kind: Options['kind']): boolean {
+  switch (kind) {
```

Comment:
We discussed this and decided not to do it because TypeScript will do the exhaustiveness checks at compile time. We could still add a default throw to be safe in some weird scenario, but we have to be careful, if implemented incorrectly it will nullify the compile time checks. See [this article](https://2ality.com/2020/02/typescript-exhaustiveness-checks-via-exceptions.html) for some ideas on how to implement it correctly.

Reply:
Wow, I wasn't aware of that. Go Typescript!